src/app/map-viewer/map-viewer.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Map } from 'leaflet';
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Rx';
+
+@Injectable()
+export class MapViewerService {
+  public map: Map;
+  public baseMaps: any;
+  public currentLat: number;
+  public currentLng:  number;
+  public altitude: number;
+  public accuracy: number;
+  hasLocation = false;
+  public center: any;
+
+  locationSource = new Subject<any>();
+
+  geoOptions = {
+      enableHighAccuracy: true,
+      maximumAge: 30000,
+      timeout: 27000
+  };
+  constructor() {
+    this.hasLocation = false;
+      if (navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition( position => {
+          this.currentLat = position.coords.latitude;
+          this.currentLng = position.coords.longitude;
+          this.altitude = position.coords.altitude;
+          this.accuracy = position.coords.accuracy;
+          this.hasLocation = true;
+
+          this.locationSource.next({
+            lat: position.coords.latitude,
+            lng: position.coords.longitude,
+            alt: position.coords.altitude,
+            acc: position.coords. accuracy,
+          });
+        }, this.error, this.geoOptions);
+      } else {
+          alert('Geolocation services are not supported by your web browser.');
+          this.hasLocation = false;
+          this.locationSource.next(false);
+      }
+  }
+  whenCurrentLocation(): Observable<any> {
+    return this.locationSource.asObservable();
+  }
+  createBaseMaps(): any {
```

Comment:
like i said above, i think this (since it's a static method) should be in the map component. we could refactor the location strategy to a different service (if we think it's necessary)